itemframework/src/item/item_scene.cpp

```diff
@@ -821,185 +822,260 @@ bool ItemScene::loadFromXml(QDomElement& dom)
-                ItemNote* note = new ItemNote();
-                note->setPos(x, y);
-
-                if (e.hasChildNodes()) {
-                    QDomElement se  = e.firstChild().toElement();
-
-                    if (se.isNull()) {
-                        qDebug() << "ItemNote has non element subtype";
-                        return false;
-                    }
-
-                    if (se.tagName() == GraphicsItemNoteDataTag) {
-                        if (!note->load(se)) {
-                            qWarning() << "Note Couldn't load it's data. (Wrong version?)";
-                        }
-                    } else {
-                        qDebug() << "Unknown subelementtype for Item_Note";
-                        return false;
-                    }
-                }
+        // Disable signals if an actual data connection is not required,
+        // for example when we are only loading these items in order to
+        // render a pixmap image of their appearance.  Simply not connecting
+        // the input and output would not work since their appearance depends
+        // on whether or not they are connected.
+        if (!shouldConnectIO) {
+            input->blockSignals(true);
+            output->blockSignals(true);
+        }
+        input->connectOutput(output); // Data connection
+
+        return loadConnector(element, input, output);
+    };
+
+    auto extractNote = [this](QDomElement const& element) -> ItemNote* {
+        auto loadNote = [](QDomElement const& element) {
+            auto note = new ItemNote();
+            qreal const x = element.attribute(XPosAttrTag).toDouble();
+            qreal const y = element.attribute(YPosAttrTag).toDouble();
+            note->setPos(x, y);
+            return note;
+        };
+
+        QScopedPointer<ItemNote> notePtr{loadNote(element)};
 
-                itemsOut->append(note);
-                connect(note, SIGNAL(changed()), this, SIGNAL(sceneRealChanged()));
+        if (!loadItemData(element, notePtr.data(), GraphicsItemNoteDataTag)) {
+            return {};
+        }
+
+        connect(notePtr.data(), SIGNAL(changed()), this, SIGNAL(sceneRealChanged()));
+        return notePtr.take();
+    };
+
+    using std::placeholders::_1;
+    using std::placeholders::_2;
+    auto reportingFunction = std::bind(&ItemScene::loadingProgress, this, _1, _2, QString{""});
+    auto const progressGoal = countChildrenIf(items, [](QDomElement const& element) {
+        return element.tagName() == GraphicsItemTag ||
+               element.tagName() == GraphicsItemNoteTag;
+    });
+
+    QHash<qint32, AbstractItem*> itemIds{};
+
+    ProgressReporter progress{reportingFunction, progressGoal, shouldReportProgress};
+    progress.report();
+
+    // since itemsOut is a pointer, it could also be captured by value, but we
+    // want to express that it is written to.
+    auto extractItem = [&itemIds, &itemsOut,
```

Comment:
you should probably swap the lambda names of `extractItem` and `extractGraphicsItem`. 
or use `extractGraphicsItem` for this one, and `extractAbstractItem` for the other one (currently named `extractGraphicsItem`